api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,33 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  const token = req.headers['x-access-token'];
+  if (!token) {
+    throw new ClientError(401, 'authentication required');
+  }
+  jwt.verify(token, process.env.TOKEN_SECRET, (err, payload) => {
+    if (err) {
+      throw err;
+    }
+    req.user = payload;
+  });
+  next();
```

Comment:
the one issue here is that you should be calling `next` within the code block for the callback to avoid a race condition. just make that quick change!